### PR TITLE
Workshop theoretical background; add evaluation and weighted loss to the workshop pipeline

### PR DESCRIPTION
**Pull request by Slowika**

### CODE REVIEW

File: setup/components/training/main.py [hunk 1]
```diff
@@ -147,17 +122,20 @@ def loss(pred: Batch) -> torch.Tensor:
...
-        batch = batch_fn(start_datetime=args.start_datetime + step * timedelta(hours=6))
+        # input for the model is the ERA5 data at start_datetime
+        inputs = batch_fn(start_datetime=args.start_datetime)
+        # target is the ERA5 data 6h in the future
+        target = batch_fn(start_datetime=args.start_datetime + timedelta(hours=6))
         optimizer.zero_grad()
-        prediction = model.forward(batch)
-        loss_value = loss(prediction)
+        prediction = model.forward(inputs)
+        loss_value = weighted_mae_loss(prediction, target)
```
Comment (Slowika) [inline setup/components/training/main.py:175]: After `dev/duncanmartyn` is updated with fixes, using `weighted_mae_loss` and deleting the loss function in this file will be the most relevant change of this PR in this file. @duncanmartyn @sali1293
